python/nano/example/pytorch/quantization/inc/resnet18_cifar_ptq.py

```diff
@@ -0,0 +1,158 @@
+    state_dict.pop('fc.weight')
+    state_dict.pop('fc.bias')
+    model.load_state_dict(state_dict, strict=False)
+    loss = nn.CrossEntropyLoss()
+    optimizer = torch.optim.SGD(model.parameters(), lr=lr, momentum=momentum,
+                                weight_decay=weight_decay)
+    metrics = [Accuracy(num_classes=num_classes, top_k=1),
+               Accuracy(num_classes=num_classes, top_k=5)]
+    return model, loss, optimizer, metrics
+
+
+def get_data_loader():
+    train_dataset = torchvision.datasets.CIFAR10("./data/train", train=True, download=True,
+                                                 transform=train_transforms)
+    train_loader = DataLoader(train_dataset, batch_size=batch_size, shuffle=True, drop_last=True)
+    test_dataset = torchvision.datasets.CIFAR10("./data/test", train=False, download=True,
+                                                transform=test_transforms)
+    test_loader = DataLoader(test_dataset, batch_size=batch_size)
+    return train_loader, test_loader
+
+
+def main():
+    # Prepare model and dataloaders
+    model, loss, optimizer, metrics = prepare_model()
+    pl_model_fp32 = Trainer.compile(model, loss, optimizer, metrics)
+    train_loader, test_loader = get_data_loader()
+    trainer = Trainer(max_epochs=num_epoches)
+
+    # Model Training. If you loaded a well-trained model, skip this
+    start = time()
+    trainer.fit(pl_model_fp32, train_loader)
+    print("Training whole time: %d s" % (time() - start))
+
+    # FP32 testing
+    start = time()
+    outputs = trainer.test(pl_model_fp32, test_loader)
+    time1 = time() - start
+    acc = outputs[0]
+    print("FP32 model size: %.3f MB" % pl_model_fp32.model_size)
+    print("Throughput: %d s" % time1)
+    print("Accuracy: top1=%.2f%% top5=%.2f %%" % (acc['test/0_Accuracy'] * 100,
+                                                  acc['test/1_Accuracy'] * 100))
+
+    # Save FP32 checkpoint
+    trainer.save_checkpoint('ResNet18_on_CIFAR_FP32_CKPT.pth')
+
+    # Run post-training quantization
+    qmodel = trainer.quantize(pl_model_fp32, calib_dataloader=train_loader,
+                              val_dataloader=test_loader,
+                              metric=Accuracy(num_classes=10, top_k=5),
+                              # recommend 'pytorch_fx' instead of 'pytorch' to enable automatic
+                              # fusion
+                              framework='pytorch_fx',
+                              approach='static',
+                              tuning_strategy='bayesian',
+                              accuracy_criterion={'higher_is_better': True, 'relative': 0.1},
+                              timeout=0,  # Limited time(seconds)
+                              max_trials=10,  # Maximum traverse to get best model
+                              raw_return=True)
+
+    # Wrap quantized model as a Pytorch-Lightning Module for testing
+    pl_model_int8 = Trainer.compile(qmodel, loss, optimizer, metrics)
```

Comment:
maybe we can never train this model anymore. do we need loss and optimizer here?